flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -1301,22 +1301,31 @@ class FMCMainDisplay extends BaseAirliners {
 
     /**
      * Any tropopause altitude up to 60,000 ft is able to be entered
-     * @param {string | number} tropo Format: NNNN or NNNNN Leading 0’s must be included. Entry is rounded to the nearest 10 ft
+     * @param {string} tropo Format: NNNN or NNNNN Leading 0’s must be included. Entry is rounded to the nearest 10 ft
      * @return {boolean} Whether tropopause could be set or not
      */
     tryUpdateTropo(tropo) {
-        const _tropo = typeof tropo === 'number' ? tropo.toString() : tropo;
-        if (_tropo.match(/^(?=(\D*\d){4,5}\D*$)/g)) {
-            const value = parseInt(_tropo.padEnd(5, '0'));
-            if (isFinite(value)) {
-                if (value >= 0 && value <= 60000) {
-                    const valueRounded = Math.round(value / 10) * 10;
-                    this.tropo = valueRounded.toString();
-                    return true;
-                }
+        if (tropo === FMCMainDisplay.clrValue) {
+            if (this.tropo) {
+                this.tropo = "";
+                return true;
             }
+            this.addNewMessage(NXSystemMessages.notAllowed);
+            return false;
         }
-        this.addNewMessage(NXSystemMessages.notAllowed);
+
+        if (!tropo.match(/^(?=(\D*\d){4,5}\D*$)/g)) {
```

Comment:
So, hm, this actually enforces 4 or 5 chars, which matches the reference:

![image](https://user-images.githubusercontent.com/929769/121057369-7d4fe100-c7bf-11eb-8452-b46966d974a2.png)

That seems to contradict the motivation for this PR?

Reply:
hmm, actually no, this is also what this pr does.

Right now the pr allows to clear a defined tropo and define the tropo with an entry inside 0-60000.
The entry will then be rounded to the nearest 10.
And the entry needs to be 4 or 5 digits long.

Reply:
Well, but doesn't the issue state

> "Any tropopause altitude up to 60,000 ft is able to be entered. The tropopause altitude cannot be entered as a flight level. Feet must be entered (i.e., do not enter 380 for FL380, the FMS interprets as 380 ft)"

...which doesn't really make sense if 380 is an invalid value.

Reply:
True, I assume Honeywell did that to prevent these issues where the pilot thought he entered a flight level but in reality he entered an altitude.

Anyways this is what the manual says...

Reply:
The Honeywell documentation contradicts itself in different sections, so that's fun.